lbz/misc.py

```diff
@@ -132,3 +133,29 @@ def deep_update(dict_to_update: dict, update_data: dict) -> None:
 
 def is_in_debug_mode() -> bool:
     return environ.get("LBZ_DEBUG_MODE", "").lower() == "true"
+
+
+def deprecated(message: str, version: str) -> Callable:
+    """
+    This is a decorator which can be used to mark functions as deprecated.
+
+    It will result in a warning being emitted when the function is used.
+    """
+
+    def decorator(func: Callable) -> Callable:
+        warnings.warn(
+            (
+                f"Call to deprecated function {func.__name__}. {message}. "
+                f"Function will be removed in {version}."
+            ),
```

Comment:
Seeing examples of the output, I think we can simplify this sentence:
```suggestion
            f"{func.__name__} - {message} (will be removed in v{version}).",
```

---
This will be shown to the user this way:
> ../lambdalizator/lbz/misc.py:xxx: DeprecationWarning: temp_function - don't use me (will be removed in v9.9.9).